pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/exchange/BlockExchange.java

```diff
@@ -100,6 +104,10 @@ public TransferableBlock send() {
     try {
       TransferableBlock block;
       long timeoutMs = _deadlineMs - System.currentTimeMillis();
+      if (_errorBlock.get() != null) {
```

Comment:
Do we need to cache the error block? Once error block is returned, there should be no more `send()`

Reply:
correct. that's why i did the same as ReceivingMailbox. let me know if this is not correct

Reply:
My question is whether we need to cache it since there will be no more call to `send()` after error block is returned the first time

Reply:
not needed. just in case we can remove actually